### DISCUSSION TITLE
db update

### DISCUSSION THREAD

**Discussion by feadrick** [Resgrid/Core#45] [Q&A]
hi shawn, i have a few question regarding the db and the version of the resgrid core.
previously im using resgrid core 0.3.25 and  i have inserted quite a lot of data inside the db.
now i have updated the the version of resgrid core to 0.4.9 , is there any potential of data loss if i run dbupdate again?

**Comment by ucswift** [comment]
Hey @feadrick! I always recommend taking a backup of the database before running DbUpdate. Most of the time it's an additive process, like adding new Tables, Columns, etc. But there could be times where we are transitioning data structures and delete old tables or columns. But when we do that we will migrate the data over to the new data structures before deleting the old.

DbUpdate also keeps track of when scripts have been run and won't re-run a script against a database. So you can run it as many times against the database as you want, it should just skip over scripts that have already been run.
